File: java/DesignPatterns/CreationalDesignPatterns/PrototypeDesignPattern/PrototypeDesignPattern.Example3/src/main/java/creational/prototype/example3/Circle.java
package creational.prototype.example3;

public class Circle extends Shape {
    public double radius;

    public Circle(String color, double x, double y, double radius) {
        super(color, x, y);
        this.radius = radius;
    }

    private Circle(Circle other) {
        super(other.color, other.x, other.y);
        this.radius = other.radius;
    }

    @Override
    public Object clone() { return new Circle(this); }

    @Override
    public void display() {
        System.out.println("Circle at (" + x + ", " + y + ") with color " + color + " and radius " + radius);
    }
}

File: nodejs/DesignPatterns/CreationalDesignPatterns/PrototypeDesignPattern/PrototypeDesignPattern.Example3/src/Circle.ts
import { Shape } from './Shape';

export class Circle extends Shape {
  radius: number;

  constructor(color: string, x: number, y: number, radius: number) {
    super(color, x, y);
    this.radius = radius;
  }

  private static fromCircle(other: Circle): Circle {
    return new Circle(other.color, other.x, other.y, other.radius);
  }

  clone(): Circle {
    return Circle.fromCircle(this);
  }

  override display(): void {
    console.log(`Circle at (${this.x}, ${this.y}) with color ${this.color} and radius ${this.radius}`);
  }
}
